criu/mount.c

```diff
@@ -3821,3 +3821,133 @@ int remount_readonly_mounts(void)
 	 */
 	return call_helper_process(ns_remount_readonly_mounts, NULL);
 }
+
+enum tree_step {
+	STEP_STOP = -2,
+	STEP_CHILD = -1,
+	STEP_SIBLING = 0,
+	STEP_PARENT = 1,
+};
+
+static struct mount_info *mnt_subtree_step(struct mount_info *mi,
+					   struct mount_info *root,
+					   enum tree_step *step)
+{
+	*step = STEP_CHILD;
+
+	if (!list_empty(&mi->children))
+		return list_entry(mi->children.next, struct mount_info, siblings);
+
+	*step = STEP_SIBLING;
+
+	while (mi->parent && mi != root) {
+		if (mi->siblings.next == &mi->parent->children) {
+			mi = mi->parent;
+			*step += STEP_PARENT;
+		} else
+			return list_entry(mi->siblings.next, struct mount_info, siblings);
+	}
+
+	*step = STEP_STOP;
+	return NULL;
+}
+
+static void mnt_resort_siblings_full(struct mount_info *tree)
+{
+	struct mount_info *m, *p;
+	LIST_HEAD(list);
+
+	pr_info("\tResorting siblings full on %d\n", tree->mnt_id);
```

Comment:
what does this mean?

Reply:
I copied these function from mnt_resort_siblings with slight change. The prototype function has the same message to tell the root of the tree we are proceeding now. May be I need to rewrite the message to `pr_info("\tResorting siblings on %d (full)\n", tree->mnt_id);` for clearness. Is it ok?

Reply:
Fixed print.